contracts/erc721/PaymentObligation.sol

```diff
@@ -105,30 +104,80 @@ contract PaymentObligation is Initializable, UserMintableERC721 {
     uint256 _tokenId,
     string memory _tokenURI,
     uint256 _anchorId,
-    bytes32 _merkleRoot,
-    string[] memory _values,
+    uint256 _nextAnchorId,
```

Comment:
do you have a plan already in the works to address this vulnerability?

Reply:
I was thinking about this. This would be a problem only if the mint fails because it's one transaction. Eth stores the failing transactions so an attacker could get the log and reserve the next_id in the anchor repo. We do not have a plan yet, just some ideas

Reply:
@pstehlik here is one idea to solve the problem:
https://github.com/centrifuge/go-centrifuge/issues/705

Reply:
@pstehlik this needs field level read permissions to be implemented efficiently. We're going to implement it then.